### PR TITLE
Refactored objectives

### PR DESCRIPTION
**Pull request by laurikskl**
Sorted out exports and imports, added the origin package to all imported functions.

Refactored to different files.

### CODE REVIEW

File: src/objectives/distance_utils.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ce::AbstractCounterfactualExplanation;
+    from::Union{Nothing,AbstractArray}=nothing,
+    agg=mean,
+    p::Real=1,
+    weights::Union{Nothing,AbstractArray}=nothing,
+)
+    if isnothing(from)
+        from = CounterfactualExplanations.factual(ce)
+    end
+    x = CounterfactualExplanations.factual(ce)
```
Comment (kmariuszk) [inline src/objectives/distance_utils.jl:16]: We're not using this x variable anywhere so it could be removed.

**Review by kmariuszk — APPROVED** [review]
There's not much to add, well done! I've managed to find only one line of code that should be removed.

File: src/objectives/Objectives.jl [hunk 2]
```diff
@@ -1,25 +1,30 @@
 module Objectives
 
 using ..CounterfactualExplanations
+using Flux
+using Flux.Losses
+using ChainRulesCore
+using ..CounterfactualExplanations
```
Comment (VincentPikand) [inline src/objectives/Objectives.jl:7]: twice

File: src/objectives/distance_utils.jl [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        ce::AbstractCounterfactualExplanation;
+        agg=mean
+    )
+
+Additional penalty for ClaPROARGenerator.
+"""
+function model_loss_penalty(ce::AbstractCounterfactualExplanation; agg=Statistics.mean)
+    x_ = CounterfactualExplanations.decode_state(ce)
+    M = ce.M
+    model = isa(M.model, Vector) ? M.model : [M.model]
```
Comment (VincentPikand) [inline src/objectives/distance_utils.jl:55]: vector is from LA package I think

**Review by VincentPikand — CHANGES_REQUESTED** [review]

File: src/objectives/distance_utils.jl (hunk 1 above)
Comment (laurikskl) [reply src/objectives/distance_utils.jl:16]: Good point

File: src/objectives/Objectives.jl [hunk 4]
```diff
@@ -1,25 +1,30 @@
 module Objectives
 
 using ..CounterfactualExplanations
+using Flux
+using Flux.Losses
```
[the remark this answers is not in the document]
Comment (laurikskl) [reply src/objectives/Objectives.jl:5]: I tried removing it for about 4 hours but decided to give up, Patrick has overwritten the functions that come from Flux.Losses and it just completely breakes when I try to remove it.

File: src/objectives/distance_utils.jl (hunk 3 above)
Comment (laurikskl) [reply src/objectives/distance_utils.jl]: Yeah added that

**Comment by VincentPikand** [comment]
do the import thingy here as well and merge dev into here to make tests pass, after that LGTM

**Pull request merged**